test/RemoteDataSource.spec.js

```diff
@@ -0,0 +1,221 @@
+          TotalRecordCount: twentyRecordsExpected.TotalRecordCount, 
+          FilteredRecordCount: twentyRecordsExpected.FilteredRecordCount,
+          TotalPages: twentyRecordsExpected.TotalPages,
+          CurrentPage: twentyRecordsExpected.CurrentPage,
+          AggregationPayload: twentyRecordsExpected.AggregationPayload
+        });
+      });
+
+      it('Should return a payload with 20 records', () => dataSource.getAllRecords(20, 0, '')
+        .then(r => {
+          expect(r.payload).to.deep.equal(twentyRecordsExpected.Payload);
+          expect(r.filteredRecordCount).to.deep.equal(twentyRecordsExpected.FilteredRecordCount);
+          expect(r.totalRecordCount).to.deep.equal(twentyRecordsExpected.TotalRecordCount);
+          expect(r.payload).to.have.lengthOf(20);
+        }));
+    });
+
+    describe('When search input is Microsoft', () => {
+      beforeEach( () => {
+        dataSource = 
+          new RemoteDataSource('http://tubular.azurewebsites.net/api/orders/paged', validColumnsSample);
+
+        mock.onPost('http://tubular.azurewebsites.net/api/orders/paged', onlyMicrosoftRecordsRequest).reply(200, {
+          Counter: onlyMicrosoftExpected.Counter,
+          Payload: onlyMicrosoftExpected.Payload,
+          TotalRecordCount: onlyMicrosoftExpected.TotalRecordCount, 
+          FilteredRecordCount: onlyMicrosoftExpected.FilteredRecordCount,
+          TotalPages: onlyMicrosoftExpected.TotalPages,
+          CurrentPage: onlyMicrosoftExpected.CurrentPage,
+          AggregationPayload: onlyMicrosoftExpected.AggregationPayload
+        });
+      });
+
+      it('Should return a payload with only Microsoft records', () => dataSource.getAllRecords(10, 0, 'Microsoft')
+        .then(r => {
+          expect(r.payload).to.deep.equal(onlyMicrosoftExpected.Payload);
+          expect(r.filteredRecordCount).to.deep.equal(onlyMicrosoftExpected.FilteredRecordCount);
+          expect(r.totalRecordCount).to.deep.equal(onlyMicrosoftExpected.TotalRecordCount);
+        }));
+    });
+
+    describe('When refresh is called', () => {
+      describe('When 10 records are requested', () => {
+        beforeEach( () => {
+          dataSource = 
+            new RemoteDataSource('http://tubular.azurewebsites.net/api/orders/paged', validColumnsSample);
+
+          mock.onPost('http://tubular.azurewebsites.net/api/orders/paged', simpleRequest).reply(200, {
+            Counter: simpleRecordsExpected.Counter,
+            Payload: simpleRecordsExpected.Payload,
+            TotalRecordCount: simpleRecordsExpected.TotalRecordCount, 
+            FilteredRecordCount: simpleRecordsExpected.FilteredRecordCount,
+            TotalPages: simpleRecordsExpected.TotalPages,
+            CurrentPage: simpleRecordsExpected.CurrentPage,
+            AggregationPayload: simpleRecordsExpected.AggregationPayload
+          });
+
+          dataSource.refresh(10, 0, '');
+        });
+
+        it('Should return a payload', done => {
```

Comment:
You cannot really confirm that the refresh method has been called this way. You should have two different payloads so you can be sure that the refresh method is actually refreshing the grid.